templates/base.html.twig

```diff
@@ -22,7 +22,7 @@
         <script>
             document.getElementById('searchField').addEventListener('input', function(event) {
             const query = event.target.value;
-            fetch(`{{ path('user_autocomplete') }}?q=` + query)
+            fetch(`https://localhost:8000/user/{{app.user.id}}/autocomplete?q=` + query)
```

Comment:
never put a `https://localhost:8000`: this will not work if your server is online ;)

try:

    fetch(`{{ path('user_autocomplete', {id: app.user.id}) }}?q=` + query)

or just:

    fetch(`{{ path('user_autocomplete') }}?q=` + query)

since you're injecting the id of the connected user, you can get it in the back end through `$this->getUser()->getId()`

Reply:
done, used first solution because autocomplete method is in UserController so Symfony needed a user id directly in the route, so I put it in the template which fetched the method